libs/template/reader.go

```diff
@@ -0,0 +1,155 @@
+package template
+
+import (
+	"context"
+	"fmt"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/databricks/cli/libs/cmdio"
+)
+
+type Reader interface {
+	// FS returns a file system that contains the template
+	// definition files. This function is NOT thread safe.
+	FS(ctx context.Context) (fs.FS, error)
+
+	// Close releases any resources associated with the reader
+	// like cleaning up temporary directories.
+	Close() error
+}
+
+type builtinReader struct {
+	name     string
+	fsCached fs.FS
+}
+
+func (r *builtinReader) FS(ctx context.Context) (fs.FS, error) {
+	// If the FS has already been loaded, return it.
```

Comment:
Looking at the call sites, it is never called multiple times. Why care?

Reply:
Removed the caching